pricegraph/src/orderbook.rs

```diff
@@ -185,6 +219,43 @@ fn as_effective_sell_price(price: &Price) -> f64 {
     FEE_FACTOR * (price.denominator as f64) / (price.numerator as f64)
 }
 
+/// A type definiton for a mapping between user IDs to user data.
+type UserMap = HashMap<UserId, User>;
+
+/// User data containing balances and number of orders.
+#[derive(Debug, Default, PartialEq)]
+struct User {
+    /// User balances per token.
+    balances: HashMap<TokenId, f64>,
+    /// The number of orders this user has.
+    num_orders: usize,
+}
+
+impl User {
+    /// Adds an encoded orderbook element to the user data, including the
+    /// reported balance and incrementing the number of orders.
+    fn include_order(&mut self, element: &Element) -> usize {
```

Comment:
naming nit: I'd find update_balance[_from_order] slightly more descriptive.

Reply:
My only issue with that name is that it doesn't capture that it also counts the orders to determine the order ID.

Reply:
Maybe `add_order` or `visit_order`? I can't think of a good name :(.

Reply:
Your point about counting orderId makes sense, I didn't catch that.